src/libasr/bwriter.h

```diff
@@ -151,6 +235,23 @@ class BinaryReader
         double *dp = (double*)p;
         return *dp;
     }
+
+    void* read_void(int64_t m_n_data, ASR::ttype_t* m_type) {
+        int kind = ASRUtils::extract_kind_from_ttype_t(m_type);
+        int n_data = m_n_data / kind;
+
+        void *p = new char[m_n_data];
+
+        for (int64_t i = 0; i < n_data; i++) {
+            uint64_t x = read_int64();
+            uint64_t *ip = &x;
+            void *p_i = (void*)((uintptr_t)p + i*kind);
+            uint64_t *p_i_64 = (uint64_t*)p_i;
+            *p_i_64 = *ip;
+        }
```

Comment:
Something like this:
```suggestion
        for (int64_t i = 0; i < n_data; i++) {
            uint8_t *p_i8 = (uint8_t*)p;
            p_i8[i] = read_int8();
        }
```